plugin/src/main/java/org/wildfly/plugin/dev/DevMojo.java

```diff
@@ -625,18 +743,32 @@ private void watch(final WatchService watcher, final DeploymentManager deploymen
                         if (result.requiresCopyResources()) {
                             triggerResources();
                         }
+                        boolean repackaged = false;
                         if (remote || result.requiresRepackage()) {
+                            // The first packaging was not an exploded war, clean it.
+                            if (requiresWarDeletion) {
+                                final Path path = resolveWarLocation();
+                                deleteRecursively(path);
+                                requiresWarDeletion = false;
+                                repackaged = true;
+                            }
```

Comment:
We should likely perform an undeploy, at least if it's not remote, before we delete the path.

Reply:
@jamezp , thank-you. I found some issues with the remote goal that I fixed. When remote is enabled, this logic shouldn't be applied and  the reprovisioning/re-start shouldn't occur. I also found an issue with the directory name used by the war plugin when the "war" goal is used. In this case, the directory can't contain the .war extension. This extension has been added for Glow to understand what kind of exploded deployment we are dealing with. In the remote case, Glow scanning is meaningless.